mlflow/transformers.py

```diff
@@ -1609,13 +1617,19 @@ def _coerce_exploded_dict_to_single_dict(self, data):
             return data
 
     def _strip_input_from_response_in_instruction_pipelines(
-        self, input_data, output, output_key, flavor_config, include_prompt
+        self,
+        input_data,
+        output,
+        output_key,
+        flavor_config,
+        include_prompt=True,
+        shrink_newlines=False,
     ):
         """
         Parse the output from instruction pipelines to conform with other text generator
         pipeline types and remove line feed characters and other confusing outputs
         """
-        replacements = {"\n\n": " "}
+        replacements = {"\n+": " ", "\\s+": " "}
```

Comment:
Is `\\s+` a newline?

Reply:
btw why do we need to replace `\\s+`? Does Dolly produce a response like `I    am Dolly`?

Reply:
If we're not declaring the match condition as a raw string (i.e., `r"\s+"`, which looks odd as a dict key), then the literal escape sequence is equivalent if using either a single `\s+` or double `\\s+`. 

example:

```
data = "Just\n\n testing\n something\n\n out\n\n\nhere.\n\n"
print("raw:")
print(data)
data = re.sub("\n+", " ", data)
print("remove newlines:")
print(data)
data_single = re.sub("\s+", " ", data)
print("remove extra spaces:")
print(data_single)
data_double = re.sub("\\s+", " ", data)
print("remove extra spaces double escape:")
print(data_double)

assert data_single == data_double
```

outputs:

```
raw:
Just

 testing
 something

 out


here.


remove newlines:
Just  testing  something  out here. 
remove extra spaces:
Just testing something out here. 
remove extra spaces double escape:
Just testing something out here. 
```